From fd4aebc2208512e8ab3b480e960d38da81908809 Mon Sep 17 00:00:00 2001
From: Kseniya Vasilchuk <vasilchukkseniia@gmail.com>
Date: Fri, 24 Apr 2015 16:02:14 +0100
Subject: [PATCH] rtspsrc: fix crashes caused by invalid caps unref

We don't get a ref to the caps returned here, so don't
unref the caps.

https://bugzilla.gnome.org/show_bug.cgi?id=748358
---
 gst/rtsp/gstrtspsrc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gst/rtsp/gstrtspsrc.c b/gst/rtsp/gstrtspsrc.c
index c545a4f..a8e5fe2 100644
--- a/gst/rtsp/gstrtspsrc.c
+++ b/gst/rtsp/gstrtspsrc.c
@@ -4457,7 +4457,6 @@ gst_rtspsrc_handle_data (GstRTSPSrc * src, GstRTSPMessage * message)
 
       if ((caps = stream_get_caps_for_pt (ostream, ostream->default_pt))) {
         gst_pad_push_event (ostream->channelpad[0], gst_event_new_caps (caps));
-        gst_caps_unref (caps);
       }
     }
     g_checksum_free (cs);
-- 
1.9.1

